### ISSUE TITLE
Is it possible to train and evaluate the model only on Nuscenes?

### ISSUE THREAD

**Issue by Jian-danai** [yuyang-cloud/Drive-OccWorld#5]
Are you combining the 3 dataset: Nuscenes, Nuscenes-Occupancy, Lyft-Level5, or did you train the model and evaluate on 3 datasets seperately?
Can I preprocess only on Nuscenes and train/evaluate only on Nuscenes? Or if you combined 3 datasets, is it possibile to train and evaluate on the combination of Nuscenes and Nuscenes-Occupancy only?

**Comment by yuyang-cloud** [comment]
To ensure a fair comparison, we train and evaluate the model separately on three distinct datasets. You can preprocess data using either the NuScenes or NuScenes-Occupancy dataset and subsequently train/evaluate on the corresponding dataset.

Regarding the E2E planning task, we conduct experiments exclusively on the NuScenes/NuScenes-Occupancy dataset, as most E2E methods report results on this benchmark. The GT occupancy constraints in the loss function are derived from the occupancy annotations provided by the NuScenes-Occupancy dataset.

**Comment by Jian-danai** [comment]
@yuyang-cloud Thanks, that's helpful! I saw all configs contain NuScenes and NuScenes-Occupancy dataset info, thus I was confused if they are used together. 
So if I just want to focus on NuScenes dataset and the E2E planning task, which config file should I use to train and evaluate the model? In other words, I want to reproduce results in Tab.5 and only downloaded nuScenes dataset.

Seems all below config files need data either from Lyft-Level5 or nuScenes-Occupancy(according to the description below), or am I understanding something incorrectly?
```
Configuration Files:

[inflated/action_condition_GMO.py](https://github.com/yuyang-cloud/Drive-OccWorld/blob/main/projects/configs/inflated/action_condition_GMO.py): Predicts General Moveable Objects (GMO) using the inflated occupancy annotations as defined in Cam4DOcc.
[inflated/action_condition_MMO.py](https://github.com/yuyang-cloud/Drive-OccWorld/blob/main/projects/configs/inflated/action_condition_MMO.py): Predicts Multi-class Moveable Objects (MMO) using the inflated occupancy annotations defined in Cam4DOcc.
[fine_grained/action_condition_GMO.py](https://github.com/yuyang-cloud/Drive-OccWorld/blob/main/projects/configs/fine_grained/action_condition_GMO.py): Predicts General Moveable Objects (GMO) using the fine-grained occupancy annotations defined in nuScenes-Occupancy.
[fine_grained/action_condition_GMO_GSO.py](https://github.com/yuyang-cloud/Drive-OccWorld/blob/main/projects/configs/fine_grained/action_condition_GMO_GSO.py): Predicts General Moveable Objects (GMO) and General Static Objects (GSO) using the fine-grained occupancy annotations defined in nuScenes-Occupancy.
[fine_grained/action_condition_MMO_MSO.py](https://github.com/yuyang-cloud/Drive-OccWorld/blob/main/projects/configs/fine_grained/action_condition_MMO_MSO.py): Predicts Multi-class Moveable Objects (MMO) and Multi-class Static Objects (MSO) using the fine-grained occupancy annotations defined in nuScenes-Occupancy.
```

**Comment by Jian-danai** [comment]
@yuyang-cloud Hi, may you advice further on which config or what modifications/steps should I use to train DriveOccWorld to verify tab.5 results(i.e., e2e planning results solely on nuScenes dataset)? Thanks.

![Image](https://github.com/user-attachments/assets/b5dfd598-9dc4-40ba-a9c8-f2600aced7f4)

**Comment by yuyang-cloud** [comment]
@Jian-danai Hi, please refer to the `fine_grained/action_condition_MMO_MSO.py` configuration. It is designed for predicting Multi-class Moveable Objects (MMO) and Multi-class Static Objects (MSO) using the fine-grained occupancy annotations provided in [nuScenes-Occupancy](https://github.com/JeffWang987/OpenOccupancy/blob/main/docs/prepare_data.md).

Please note the following:
- The nuScenes-Occupancy dataset is derived from nuScenes and includes voxel-wise occupancy annotations for each frame.

- `MMO+MSO` provides a more detailed representation of occupancy states, covering both dynamic objects and static elements (e.g., drivable areas), which are used as cost conditions in our E2E planner.

- You can pretrain the model using the `fine_grained/action_condition_MMO_MSO.py` configuration, then set `turn_on_plan = True`, and utilize [PlanHead_v1(BaseModule)](https://github.com/yuyang-cloud/Drive-OccWorld/blob/fdbca0e51f882459fa088649a6296b388b55d5b1/projects/mmdet3d_plugin/bevformer/dense_heads/plan_head.py#L150) for the E2E planning task.

**Comment by Jian-danai** [comment]
@yuyang-cloud Thanks for the clear answer. I trained the model with the original `fine_grained/action_condition_MMO_MSO.py` config and utilized both nuscenes and nuscenes-occupancy dataset.

Before the final E2E task evaluation, Do I need a finetune after setting `turn_on_plan = True` and adding the `plan_head` as mentioned there?
https://github.com/yuyang-cloud/Drive-OccWorld/issues/6#issuecomment-2804057141

**Comment by yuyang-cloud** [comment]
Yes, you need to train the `plan_head` for the E2E task. Both training from scratch and fine-tuning are effective, although fine-tuning may yield better results.

**Comment by Jian-danai** [comment]
@yuyang-cloud Hi, I trained fine_grained_mmo_mso with planner enabled, the only log that matches planning metrics is shown below. I don't see the result of 3s which was reported in the paper, do u have other settings in metric calculatation to reproduce your paper results in tab5?
how to calculate with the following three settings
- NoAvg evaluation protocol
- TemAvg protocol
- enable the use of ego status in the planning module 

```
|   metrics   |  0.5s  |  1.0s  |  1.5s  |  2.0s  |  2.5s  |
+-------------+--------+--------+--------+--------+--------+
|   obj_col   | 0.0000 | 0.0000 | 0.0000 | 0.0000 | 0.0000 |
| obj_box_col | 0.0008 | 0.0014 | 0.0008 | 0.0012 | 0.0016 |
|      L2     | 0.1451 | 0.1626 | 0.2428 | 0.3235 | 0.3940 |
+-------------+--------+--------+--------+--------+--------+
```

**Comment by Kivw** [comment]
> [@yuyang-cloud](https://github.com/yuyang-cloud) Hi, I trained fine_grained_mmo_mso with planner enabled from scratch, the only log that matches planning metrics is shown below. I don't see the result of 3s which was reported in the paper, do u have other settings in metric calculatation to reproduce your paper results in tab5? how to calculate with the following three settings
> 
> * NoAvg evaluation protocol
> * TemAvg protocol
> * enable the use of ego status in the planning module
> 
> ```
> |   metrics   |  0.5s  |  1.0s  |  1.5s  |  2.0s  |  2.5s  |
> +-------------+--------+--------+--------+--------+--------+
> |   obj_col   | 0.0000 | 0.0000 | 0.0000 | 0.0000 | 0.0000 |
> | obj_box_col | 0.0008 | 0.0014 | 0.0008 | 0.0012 | 0.0016 |
> |      L2     | 0.1451 | 0.1626 | 0.2428 | 0.3235 | 0.3940 |
> +-------------+--------+--------+--------+--------+--------+
> ```

Hello, can the results reported in the paper be reproduced? I want to focus on autonomous driving based on world models. Is this work a good benchmark? Do you have any recommendations?
